src/vehicle.h

```diff
@@ -858,6 +858,23 @@ class vehicle : public JsonSerializer, public JsonDeserializer
 
     /*@}*/
 
+    /**
+     * If target for the turret is set, it returns the set target and unsets target.
+     * Otherwise finds a target and returns it.
+     * @return A target for the turret or turret's position if no target is found.
+     */
+    tripoint turret_acquire_target( vehicle_part &pt );
+    /**
+     * Fires a loaded turret at the target.
+     * @return Number of shots fired.
+     */
+    int turret_fire( vehicle_part &pt, const tripoint &targ );
+
+    void turret_reload( vehicle_part &pt );
```

Comment:
> assuming everything works, haven't checked that

I think it currently doesn't.

> but I can't get it to load a CARGO turret for some reason

This is one of the nastier parts of the current implementation. It prevents for example having a turret and seat on the same tile. I'm fairly close* to completing an implementation that depends on #17732 and closes #17618 and should hopefully resolve @Coolthulhu's existing issue

`*` it works but the code needs some cleanup